content/en/blog/_posts/2020-03-30-two-phased-canary-rollout-with-gloo.md

```diff
@@ -0,0 +1,772 @@
+To simulate a load test, let's shift half the traffic to `v2`:
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/3-progressive-traffic-shift-to-v2/load-test.png)
+
+This can be expressed on our virtual service by adjusting the weights:
+
+```yaml
+apiVersion: gateway.solo.io/v1
+kind: VirtualService
+metadata:
+  name: echo
+  namespace: gloo-system
+spec:
+  virtualHost:
+    domains:
+      - '*'
+    routes:
+      - matchers:
+          - headers:
+              - name: stage
+                value: canary
+            prefix: /
+        routeAction:
+          single:
+            upstream:
+              name: echo
+              namespace: gloo-system
+            subset:
+              values:
+                version: v2
+      - matchers:
+          - prefix: /
+        routeAction:
+          multi:
+            destinations:
+              - destination:
+                  upstream:
+                    name: echo
+                    namespace: gloo-system
+                  subset:
+                    values:
+                      version: v1
+                # Update the weight so 50% of the traffic hits v1
+                weight: 50
+              - destination:
+                  upstream:
+                    name: echo
+                    namespace: gloo-system
+                  subset:
+                    values:
+                      version: v2
+                # And 50% is routed to v2
+                weight: 50
+```
+
+We can apply this to the cluster with the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/3-progressive-traffic-shift-to-v2/vs-2.yaml
+```
+
+Now when we send traffic to the gateway, we should see half of the requests return `version:v1` and the 
+other half return `version:v2`. 
+
+```bash
+➜ curl $(glooctl proxy url)/
+version:v1
+➜ curl $(glooctl proxy url)/
+version:v2
+➜ curl $(glooctl proxy url)/
+version:v1
+```
+
+In practice, during this process it's likely you'll be monitoring some performance and business metrics 
+to ensure the traffic shift isn't resulting in a decline in the overall quality of service. We can even 
+leverage operators like [Flagger](https://github.com/weaveworks/flagger) to help automate this Gloo 
+workflow. Gloo Enterprise integrates with your metrics backend and provides out of the box and dynamic, 
+upstream-based dashboards that can be used to monitor the health of the rollout. 
+We will save these topics for a future post on advanced canary testing use cases with Gloo. 
+
+### Finishing the rollout
+
+We will continue adjusting weights until eventually, all of the traffic is now being routed to `v2`:
+
+![](https://raw.githubusercontent.com/solo-io/gloo-ref-arch/blog-30-mar-20/platform/prog-delivery/two-phased-with-os-gloo/3-progressive-traffic-shift-to-v2/final-shift.png)
```

Comment:
1. This image should have an alt attribute.
2. Can you copy the image into the repo so that it's served via the content CDN and the image is in source control?